xgboost/gbt.py

```diff
@@ -135,9 +137,13 @@ def fit():
         dtrain = xgb.DMatrix(X_train, y_train)
         return xgb.train(xgb_params, dtrain, params.n_estimators)
 
-    def predict():
-        dtest = xgb.DMatrix(X_test, y_test)
-        return booster.predict(dtest)
+    if params.inplace_predict == False:
+        def predict():
+            dtest = xgb.DMatrix(X_test, y_test)
+            return booster.predict(dtest)
+    else:
+        def predict():
+            return booster.inplace_predict(np.ascontiguousarray(X_test.values, dtype=np.float32))
```

Comment:
Why np.ascontiguousarray?

Reply:
inplace supports only numpy c_contigous array

Reply:
I hope Sklearn API will support this method soon for pandas and numpy.

Reply:
And float32 only?

Reply:
If data is `float64` same conversion will be performed inside `inplace_predict`
So as a conversion is included to measurement I don't think that there is any problem